proto/scenario.proto

```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
+package proto;
+
+message Scenario {
+  Load load = 1;
+  Cluster cluster = 2;
+  Workload workload = 3;
+  repeated Plugin plugin = 4;
+}
+
+message Load {
+  oneof arrival_rate_oneof {
+    LoadConstant constant = 1;
+    LoadPeriodic periodic = 3;
+    LoadCustom custom = 3;
+  }
+  oneof request_weight_oneof {
+    RequestConstant constant = 4;
+  }
+}
+
+message LoadConstant {
+  int32 arrivals_per_second = 1;
+}
+
+message LoadPeriodic {
+  oneof shape_oneof {
+    ShapeSine sine = 1;
+    ShapeAsdr asdr = 2;
+  }
+}
+
+message ShapeSine {
+  LoadInterval interval = 1;
+}
+
+message ShapeAsdr {
+  LoadInterval initial = 1;
+  LoadInterval attack = 2;
+  LoadInterval sustain = 3;
+  LoadInterval decay = 4;
+  LoadInterval release = 5;
+}
+
+message LoadCustom {
+  repeated LoadInterval interval = 1;
+}
+
+message LoadInterval {
+  int32 width_milliseconds = 1;
+  int32 amplitude_aps = 2;
+}
+
+message RequestConstant {
+  int32 cpu_milliseconds = 1;
+  int32 io_milliseconds = 2;
+  int32 memory_bytes = 3;
+}
+
+message Cluster {
+  // infinite capacity
+}
+
+message Workload {
+  int32 readiness_delay_milliseconds = 1;
+  int32 termination_delay_milliseconds = 2;
+}
+
+message Plugin {
+  string type = 1;
+  int32 version = 2;
+  PluginPoint point = 3;
+  Any configuration = 4;
+}
+
+enum PluginPoint {
+  HORIZONTAL_AUTOSCALING = 0;
+  VERTICAL_AUTOSCALING = 1;
+  REQUEST_ROUTING = 2;
+}
+
+message Result {
+  repeated ResultMetric metric = 1;
+}
+
+message Metric {
+  MetricType type = 1;
+  MetricResource resource = 2
+  repeated MetricPoint point = 3;
+}
+
+enum MetricType {
+  COUNT = 0;
+  AVERAGE = 1;
```

Comment:
Since we also have `P50`, I suggest calling this `MEAN`.

We should also add `P25` (for interquartile range) and `POP_STDDEV` (the population standard deviation, since this is a total simulation).

Reply:
Good idea.  I'll make it MEAN.  And add P25 and POP_STDDEV.